### PR TITLE
{WIP} Upgrade to 2.12

### PR DESCRIPTION
**Pull request by sritchie**
Before we can finish this out we'll need to

- [ ] publish Algebird 0.13.0 and the latest Chill
- [ ] Use the `sbt-doge` plugin to turn off 2.12 cross-builds for anything that touches Scrooge, which doesn't yet have a 2.12 build

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -279,11 +277,11 @@ lazy val noPublishSettings = Seq(
...
  */
-val unreleasedModules = Set[String]("hadoop-test") //releases 0.11
+val ignoredModules = Set[String]("benchmarks")
 
 def youngestForwardCompatible(subProj: String) =
   Some(subProj)
-    .filterNot(unreleasedModules.contains(_))
+    .filterNot(ignoredModules.contains(_))
     .map {
     s => "com.twitter" %% (s"scalding-$s") % "0.16.0"
```
Comment (johnynek) [inline build.sbt:286]: I thought a later version had been released.

We should write code to just compute the previous version if the current version does not end in 0

Comment (sritchie) [reply build.sbt:286]: No later version's out, according to Maven Central.

**Comment by piyushnarang** [comment]
@sritchie do you have the time to pick this back up? We've published algebird 0.13.0, and chill 0.7.7 (which depends on algebird 0.13.0). If you're busy, I can use this as a base and spin up a new PR. Once we have this merged we can proceed with cutting a new Scalding 0.17.0 release (https://github.com/twitter/scalding/pull/1641)

**Comment by piyushnarang** [comment]
Changes are here in case you want to build on them: https://github.com/piyushnarang/scalding/tree/pnarang/scala212wip

**Comment by johnynek** [comment]
closed by #1646

**Pull request closed without merging**